### ISSUE TITLE
Add own relay (coturn) server

### ISSUE THREAD

**Issue by vampywiz17** [netbirdio/netbird#957]
**Is your feature request related to a problem? Please describe.**
it would be good that use a own coturn/relay server. It contain the self-hosted bundle, but somtime enought the free plan, but would be good a fast/self-hosted relay.

**Describe the solution you'd like**
Create a option in app to use own coturn/relay server and make a note, how to install a own one.

**Comment by vampywiz17** [comment]
This thing is planned? 

Now i use Tailscale + own Derper to fastest access to my home network, every scenario. (p2p are priority and derper, if the remote network too restricted)

I would like to switch netbird, mainly because of kernel wireguard.

**Comment by mlsmaycon** [comment]
Hello @vampywiz17 it is planned for 2024, we don't have yet a proper estimation in which quarter we will be working on it, but I will let you know as soon as we plan it.

**Comment by kuyoeku** [comment]
Is it possible to add a relay mode to the peers? Each peer has its own algorithm to determine which relay peer can achieve the lowest latency to reach the target peer.

**Comment by remann2** [comment]
any updates?

**Comment by 1nerdyguy** [comment]
Assume this can be closed out with the new relays?

**Comment by vampywiz17** [comment]
> Assume this can be closed out with the new relays?

You talking about these, right?

> Coming Soon
New relay system
We are moving away from the TURN relay (coturn) to our own relay implementation based on WebSocket. This new system will ensure that all relayed connections utilize a single secure port (443), instead of allocating one port per connection as before. The update is expected to enhance performance and ensure faster, more reliable connections, even in stricter network environments. 🚀 That's a wrap for this month's newsletter. Thank you for being part of our community. We appreciate your support and feedback. Until next time!

But it mean that your provided relay will change working method, not self hosted relay. If i understand it correctly. Anyway, it sound good, nof i finally able to replace my tailscale network (now netbird not work some cases, for example some heavy firewalled public networks...)

**Comment by 1nerdyguy** [comment]
The new method allows you to deploy the relays independently of the management service, as it's now a stand alone docker instance. You can also add them to the management.json file, and have multiple, self-hosted relays. 

I've tested this myself due to similar issues. Hosting 5x relays by just spinning up the docker container on a host by each heavily firewallwed area, forwarding 1 port, and updating the management.json. 

Works great!

**Comment by vampywiz17** [comment]
@1nerdyguy 

One question left :)

If i do a cloud service (free tear, enought to me) and also install a new relay on my infra, it possible to add it to my cloud settting?

(i not want to provide a full netbird, only relay)

**Comment by 1nerdyguy** [comment]
No clue on that, but in this case you shouldn't need to, as you're able to use the relays of the cloud environment. 

Are you stating you'd like to use the free-tier of Netbird, you're getting Managed netbird there. That includes, according to their docs: https://docs.netbird.io/selfhosted/self-hosted-vs-cloud-netbird, geo-distributed relays.

**Comment by vampywiz17** [comment]
> No clue on that, but in this case you shouldn't need to, as you're able to use the relays of the cloud environment.
> 
> Are you stating you'd like to use the free-tier of Netbird, you're getting Managed netbird there. That includes, according to their docs: https://docs.netbird.io/selfhosted/self-hosted-vs-cloud-netbird, geo-distributed relays.

Yes, i know, that i able to use cloud based relays. but some cases (for example speed ) it better to use own relay.

For example:

Current way:

Client 1 (at home) ----->   Cloud Relay (for example a different country, it use other people, etc...)  ----->   Client 2 (a different place)

Preferred way: 

Client 1 (at home) ----->   my own hosted Relay (same network that my client 1 , essentially a "other" client with TLS and https that accept all firewall )  ----->   Client 2 (a different place)

**Comment by 1nerdyguy** [comment]
Nope, I get what you're saying. 

I'm saying that, to my knowledge, that's not an option. So guess you're right, it's still a thing, but you we need to tweak it to include the new relays 

That being said, it's no longer a issue, as much as a feature request.

**Comment by vampywiz17** [comment]
these topic start with " feature request" not issue.

the original post:

> it would be good that use a own coturn/relay server. It contain the self-hosted bundle, b**ut somtime enought the free plan**, but would be good a fast/self-hosted relay.

But i found these in release note on v0.29.0

> Cloud support for the new relay feature is coming soon*.

Not clear to me that mean:

1. Cloud relays will update now relay type
2. Cloud will able to configure your own relay server

Of course that 2. are the best to me. (and the answer these feature request)

**Comment by kmahyyg** [comment]
Same problem here on my side, since Netbird managed cloud relay is blocked/very slow on my side, I'd like to add a self-hosted relay server in my managed netbird instance, feature request +1

**Comment by lonnietc** [comment]
Hi All,  

I also want to add relays, but outside of Docker and have compiled the netbird relay Golang code into binaries for Windows, Linux, and MacOS, but just cannot seem to find any documentation on how to run it, or also how to get a list of active network relays from the command line.  As a node, I have also set up the self-hosting netbird for testing all of this, and it runs nicely with the basic install.

Any thoughts on running and listing these non-docker relays would be truly appreciated.

**Comment by WXZhao7** [comment]
I noticed this post, which shows that adding extra relays for self-hosted deployments isn’t too difficult. 

[Reddit - Hosting Netbird with multiple relays](https://www.reddit.com/r/selfhosted/comments/1im3z96/hosting_netbird_with_multiple_relays/)

It might improve network reliability if the dashboard → settings menu allowed adding more relays or only using self-hosted relays.

**Comment by PizzaLovingNerd** [comment]
Hello, we are closing this aas a stale issue. For self-hosted deployments, custom relay servers are now supported via the standalone netbirdio/relay image and config.yaml, see [Set Up External Relay Servers](https://docs.netbird.io/selfhosted/maintenance/scaling/set-up-external-relays). The old coturn-based relay path has been replaced by the WebSocket relay system ([#2566](https://github.com/netbirdio/netbird/issues/2566)).

**Comment by vampywiz17** [comment]
@PizzaLovingNerd  but it only work with self hosted netbird right? the original question are that possible to add own relay server to cloud tier. (like tailscale DARP)

**Comment by TechHutTV** [comment]
@vampywiz17 @WXZhao7 @Uskrai I double checked with the Dev team. This feature is planned but there is no ETA.
